tools/ubuntu-setup/docker.sh

```diff
@@ -14,7 +14,7 @@ sudo apt-cache policy docker-engine
 sudo apt-get -y install linux-image-extra-$(uname -r)
 
 # DOCKER
-sudo apt-get install -y --force-yes docker-engine=1.12.0-0~trusty
+sudo apt-get install -y --force-yes docker-engine
```

Comment:
This is a bit dangerous. While OpenWhisk works on newer docker versions perfectly fine, it is not CIed as of today. We got bitten by docker version regressions so we very carefully update and today recommend using 1.12 for anything close to a production environment.

Reply:
The reason I did that was in order to bring the latest stable version provided by the defined repository for a specific Ubuntu OS.
Docker seems to change really frequently and if I remember well on the Ubuntu 16.04 VM that I made the tests that specific version wasn't available at that time in the repository only a newer one (docker-engine 17.05) .

I remade just know a test on Ubuntu 16.04 and it seems that the key repo isn't working anymore at least for Ubuntu 16.04, haven't tested on a Ubuntu 14.04 server yet.
According to the docker documentation for Ubuntu trusty/xenial and later the docker-engine package seems to have become docker-ce. https://docs.docker.com/engine/installation/linux/docker-ce/ubuntu/ 
Will come back with an update for this script if that is ok with you.

Reply:
Also related to https://github.com/apache/incubator-openwhisk/pull/2947#issuecomment-343937990 @csantanapr